spec/puppet-syntax/manifests_spec.rb

```diff
@@ -10,50 +10,85 @@
 
-    expect(res.size).to eq(1)
-    expect(res[0]).to match(/Syntax error at .*:3$/)
+    expect(output.size).to eq(1)
+    expect(output[0]).to match(/Syntax error at .*:3$/)
+    expect(has_errors).to eq(true)
   end
 
   it 'should return a warning from an invalid file' do
     files = fixture_manifests('fail_warning.pp')
-    res = subject.check(files)
+    output, has_errors = subject.check(files)
 
-    expect(res.size).to eq(2)
-    expect(res[0]).to match(/Unrecognised escape sequence '\\\[' .* at line 3$/)
-    expect(res[1]).to match(/Unrecognised escape sequence '\\\]' .* at line 3$/)
+    expect(output.size).to eq(2)
+    expect(output[0]).to match(/Unrecognised escape sequence '\\\[' .* at line 3$/)
+    expect(output[1]).to match(/Unrecognised escape sequence '\\\]' .* at line 3$/)
+    expect(has_errors).to eq(true)
   end
 
   it 'should ignore warnings about storeconfigs' do
     files = fixture_manifests('pass_storeconfigs.pp')
-    res = subject.check(files)
+    output, has_errors = subject.check(files)
+
+    expect(output).to eq([])
+    expect(has_errors).to eq(false)
 
-    expect(res).to eq([])
   end
 
   it 'should read more than one valid file' do
     files = fixture_manifests(['pass.pp', 'pass_storeconfigs.pp'])
-    res = subject.check(files)
+    output, has_errors = subject.check(files)
 
-    expect(res).to eq([])
+    expect(output).to eq([])
+    expect(has_errors).to eq(false)
   end
 
   it 'should continue after finding an error in the first file' do
     files = fixture_manifests(['fail_error.pp', 'fail_warning.pp'])
-    res = subject.check(files)
+    output, has_errors = subject.check(files)
+
+    expect(output.size).to eq(3)
+    expect(output[0]).to match(/Syntax error at '\}' .*:3$/)
+    expect(output[1]).to match(/Unrecognised escape sequence '\\\[' .* at line 3$/)
+    expect(output[2]).to match(/Unrecognised escape sequence '\\\]' .* at line 3$/)
+    expect(has_errors).to eq(true)
+  end
 
-    expect(res.size).to eq(3)
-    expect(res[0]).to match(/Syntax error at '\}' .*:3$/)
-    expect(res[1]).to match(/Unrecognised escape sequence '\\\[' .* at line 3$/)
-    expect(res[2]).to match(/Unrecognised escape sequence '\\\]' .* at line 3$/)
+  describe 'deprecation notices' do
+    # These tests should fail in Puppet 4, but we need to wait for the release
+    # before we'll know exactly how to test it.
+    if Puppet::Util::Package.versioncmp(Puppet.version, '3.7') >= 0
+      context 'on puppet >= 3.7' do
+        it 'should return deprecation notices as warnings' do
+          files = fixture_manifests('deprecation_notice.pp')
+          output, has_errors = subject.check(files)
+
+          expect(has_errors).to eq(false)
+          expect(output.size).to eq(2)
+          expect(output[0]).to match(/The use of 'import' is deprecated/)
+          expect(output[1]).to match(/Deprecation notice:/)
+        end
+      end
+    elsif Puppet::Util::Package.versioncmp(Puppet.version, '3.5') < 0
```

Comment:
The tests here don't included Puppet 3.6; can you please expand these tests or add new tests to verify that behaviour?